mPower2/mPower2/View Controllers/TaskBrowserViewController.swift

```diff
@@ -0,0 +1,366 @@
+//  mPower2
+//
+//  Copyright © 2018 Sage Bionetworks. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without modification,
+// are permitted provided that the following conditions are met:
+//
+// 1.  Redistributions of source code must retain the above copyright notice, this
+// list of conditions and the following disclaimer.
+//
+// 2.  Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation and/or
+// other materials provided with the distribution.
+//
+// 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission. No license is granted to the trademarks of
+// the copyright holders even if such marks are included in this software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+import UIKit
+import Research
+import MotorControl
+
+protocol TaskBrowserViewControllerDelegate {
+    func taskBrowserViewControllerToggleVisibility()
+}
+
+class TaskBrowserViewController: UIViewController, RSDTaskViewControllerDelegate, TaskBrowserTabViewDelegate {
+    
+    // Used by our potential parent VC to show/hide our view
+    class func tabsHeight() -> CGFloat {
+        return 50.0
+    }
+    
+    private let kCollectionCellIdentifier = "TaskCollectionViewCell"
+    private let kTopInset: CGFloat = 30.0
+
+    // TODO: jbruhin 5-1-18 need to optimize for all screen sizes
+    open var cellSize = CGSize(width: 80.0, height: 80.0)
+    open var minCellSpacing: CGFloat = 30.0
+    
+    public var shouldShowTopShadow = true
+    public var taskGroups: [RSDTaskGroup]?
+
+    private var collectionView: UICollectionView!
+    private var tabButtonStackView: UIStackView!
+    private var selectedTaskGroup: RSDTaskGroup?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupView()
+    }
+    
+    func shouldShowTabs() -> Bool {
+        guard let taskGroups = taskGroups else {
+            return false
+        }
+        return taskGroups.count > 1
+    }
+    
+    func setupView() {
```

Comment:
Can you please use a storyboard? That makes it easier to visualize the constraints and tweek things for different iPhone sizes. Thanks

Reply:
All the content here - including the 'Tracking' and 'Measuring' tabs - is generated dynamically based on the model. The view controller is assigned an array of RSDTaskGroups. A tab is created for each one of the task groups and the content for each task group is derived from the tasks associated with the group. So...I'm not sure what would go into a storyboard. Without the model, which I can't assign in the storyboard, there would be no content.